components/SerialServer/src/serial.c

```diff
@@ -24,7 +24,8 @@
 #include "server_virtqueue.h"
 
 #define ESCAPE_CHAR '@'
-#define MAX_CLIENTS 12
+#define MAX_CLIENTS 10
+#define MAX_BADGE   (MAX_CLIENTS-1)
```

Comment:
A badge value of 0 is actually the same as unbadged and so shouldn't be used as the client could make copies of the cap and give them any badge.

Reply:
That would cause issues with this PR: https://github.com/seL4/camkes-vm/pull/40

How do you suggest we configure the badges/client relationship then?

Reply:
Currently, there's a different ID assignment for each of the interfaces, getchar, putchar and batch.  Making this ID assignment unified requires either making a single interface or making the system designer assign IDs across all of the connection instances manually. It used to be the second way but it was easy to mis-configure and led to lots of boilerplate. 

Maybe you just want to be able to assign a client_id to the getchar interface and don't change the ID assignment for the putchar/batch interfaces.